runtime/j9vm/jvm.c

```diff
@@ -322,6 +322,351 @@ static BOOLEAN librariesLoaded(void);
+#define J9_SIGPWR SIGPWR
+#else /* defined(SIGPWR) */
+#define J9_SIGPWR J9_SIG_ERR
+#endif /* defined(SIGPWR) */
+
+#if defined(SIGUSR1)
+#define J9_SIGUSR1 SIGUSR1
+#else /* defined(SIGUSR1) */
+#define J9_SIGUSR1 J9_SIG_ERR
+#endif /* defined(SIGUSR1) */
+
+#if defined(SIGUSR2)
+#define J9_SIGUSR2 SIGUSR2
+#else /* defined(SIGUSR2) */
+#define J9_SIGUSR2 J9_SIG_ERR
+#endif /* defined(SIGUSR2) */
+
+#if defined(SIGPROF)
+#define J9_SIGPROF SIGPROF
+#else /* defined(SIGPROF) */
+#define J9_SIGPROF J9_SIG_ERR
+#endif /* defined(SIGPROF) */
+
+#if defined(SIGDANGER)
+#define J9_SIGDANGER SIGDANGER
+#else /* defined(SIGDANGER) */
+#define J9_SIGDANGER J9_SIG_ERR
+#endif /* defined(SIGDANGER) */
+
+#if defined(SIGVTALRM)
+#define J9_SIGVTALRM SIGVTALRM
+#else /* defined(SIGVTALRM) */
+#define J9_SIGVTALRM J9_SIG_ERR
+#endif /* defined(SIGVTALRM) */
+
+#if defined(SIGMIGRATE)
+#define J9_SIGMIGRATE SIGMIGRATE
+#else /* defined(SIGMIGRATE) */
+#define J9_SIGMIGRATE J9_SIG_ERR
+#endif /* defined(SIGMIGRATE) */
+
+#if defined(SIGPRE)
+#define J9_SIGPRE SIGPRE
+#else /* defined(SIGPRE) */
+#define J9_SIGPRE J9_SIG_ERR
+#endif /* defined(SIGPRE) */
+
+#if defined(SIGCPUFAIL)
+#define J9_SIGCPUFAIL SIGCPUFAIL
+#else /* defined(SIGCPUFAIL) */
+#define J9_SIGCPUFAIL J9_SIG_ERR
+#endif /* defined(SIGCPUFAIL) */
+
+#if defined(SIGGRANT)
+#define J9_SIGGRANT SIGGRANT
+#else /* defined(SIGGRANT) */
+#define J9_SIGGRANT J9_SIG_ERR
+#endif /* defined(SIGGRANT) */
+
+#if defined(SIGRETRACT)
+#define J9_SIGRETRACT SIGRETRACT
+#else /* defined(SIGRETRACT) */
+#define J9_SIGRETRACT J9_SIG_ERR
+#endif /* defined(SIGRETRACT) */
+
+#if defined(SIGSOUND)
+#define J9_SIGSOUND SIGSOUND
+#else /* defined(SIGSOUND) */
+#define J9_SIGSOUND J9_SIG_ERR
+#endif /* defined(SIGSOUND) */
+
+#if defined(SIGSAK)
+#define J9_SIGSAK SIGSAK
+#else /* defined(SIGSAK) */
+#define J9_SIGSAK J9_SIG_ERR
+#endif /* defined(SIGSAK) */
+
+#if defined(SIGPOLL)
+#define J9_SIGPOLL SIGPOLL
+#else /* defined(SIGPOLL) */
+#define J9_SIGPOLL J9_SIG_ERR
+#endif /* defined(SIGPOLL) */
+
+#if defined(SIGINFO)
+#define J9_SIGINFO SIGINFO
+#else /* defined(SIGINFO) */
+#define J9_SIGINFO J9_SIG_ERR
+#endif /* defined(SIGINFO) */
+
+#if defined(SIGSTKFLT)
+#define J9_SIGSTKFLT SIGSTKFLT
+#else /* defined(SIGSTKFLT) */
+#define J9_SIGSTKFLT J9_SIG_ERR
+#endif /* defined(SIGSTKFLT) */
+
+#if defined(SIGBREAK)
+#define J9_SIGBREAK SIGBREAK
+#else /* defined(SIGBREAK) */
+#define J9_SIGBREAK J9_SIG_ERR
+#endif /* defined(SIGBREAK) */
+
+#if defined(SIGUNUSED)
+#define J9_SIGUNUSED SIGUNUSED
+#else /* defined(SIGUNUSED) */
+#define J9_SIGUNUSED J9_SIG_ERR
+#endif /* defined(SIGUNUSED) */
+
+#define J9_IS_SIGNAL_SPECIAL(SIG_NUM) ((SIG_NUM != J9_SIG_ERR) && ((SIG_NUM == J9_SIGQUIT) || (SIG_NUM == J9_SIGHUP) || (SIG_NUM == J9_SIGINT) || (SIG_NUM == J9_SIGTERM)))
+#define J9_IS_SIGNAL_USED_BY_VM(SIG_NUM) ((SIG_NUM != J9_SIG_ERR) && ((SIG_NUM == J9_SIGFPE) || (SIG_NUM == J9_SIGILL) || (SIG_NUM == J9_SIGSEGV)))
```

Comment:
The uses of `SIG_NUM` in the expansion of the macro should be enclosed in parentheses.
It may make sense to convert these macros to static helper functions so the tests for signals can be conditional on whether the corresponding signal is defined.

Reply:
fixed - converted to static helper functions.